pkgs/by-name/bc/bcache-tools/package.nix

```diff
@@ -28,34 +26,31 @@ stdenv.mkDerivation rec {
 
   doInstallCheck = true;
 
-  # * Remove broken install rules (they ignore $PREFIX) for stuff we don't need
-  #   anyway (it's distro specific stuff).
-  # * Fixup absolute path to modprobe.
   prePatch = ''
+    # * Remove distro specific install rules which are not used in NixOS.
+    # * Remove binaries for udev which are not needed on modern systems.
     sed -e "/INSTALL.*initramfs\/hook/d" \
         -e "/INSTALL.*initcpio\/install/d" \
         -e "/INSTALL.*dracut\/module-setup.sh/d" \
-        -e "s/pkg-config/$PKG_CONFIG/" \
+        -e "/INSTALL.*probe-bcache/d" \
         -i Makefile
+    # * Remove probe-bcache which is handled by util-linux
+    # * Replace bcache-register binary with a write to sysfs
+    sed -e "/probe-bcache/d" \
+        -e "s@.*bcache-register.*@RUN+=\"${bash}/bin/sh -c 'echo \$tempnode > /sys/fs/bcache/register'\"@" \
```

Comment:
Does substituteInPlace work here? If so, I think it's preferable, since it has the --replace-fail flag (which detect breakage at build time).